### ISSUE TITLE
Running tests inside Jenkins which is inside docker container

### ISSUE THREAD

**Issue by skisel** [zalando/zalenium#136]
I have a problem with zalenium:

We are running jenkins inside docker container as well as zalenium with docker-selenium. Jenkins network is not linked to zalenium containers (could be solved however). So I run my tests inside local docker network (172.17.0.1:4444), however it seems my tests cannot get connected to selenium node, because in the configuration it is bound to localhost, so it tries to connect to localhost, here are my logs from the tests:

[19:09:25] I/hosted - Using the selenium server at http://172.17.0.1:4444/wd/hub
[19:11:16] E/launcher - cannot forward the request localhost:40002 failed to respond
[19:11:16] E/launcher - WebDriverError: cannot forward the request localhost:40002 failed to respond

Everything works if I run tests on the host server, not inside container. I assume it wont work if I run zalenium on server1 and then on server2 try to run my tests, as there is no way to configure nodes (to make them non-localhost).

Is there any other way to fix the problem than linking containers or tunneling networks?

**Comment by skisel** [comment]
I tried to use zalenium on one server and run the tests on the other - it does not work. I think localhost somehow is hardcoded somewhere

**Comment by elgalu** [comment]
Hi, if you want to run tests on a different server than Zalenium just configure your tests properly.

```
browser.get("http://your-test-server-ip/");
```

**Comment by diemol** [comment]
Hi @skisel,

Were you able to make it work?

If not, some additional questions...

What do you use when you create the RemoteWebDriver object? What url? 

If the tests are inside a container as well, you might need to use the IP of the container where Zalenium is running. 

Another idea would be to start everything via docker-compose, so everything is linked.

**Comment by skisel** [comment]
Hi @diemol ,

Thank you for the reply!

We use protractor, so we do not use selenium itself.

Actually we made it work, but I do not have explanation yet why I saw the message I paste you. We did two changes: one in the configuration of jenkins docker container (we add --net=host) which did not help to solve the problem (still same issue) and then it turned out the server was running with no internet connection and url was an outside one, so it could not connect. 

I will revert (--net=host) configuration and will let you know if it still works (I assume it will), probably just message was misleading.
